src/clients/UBAClient/UBAClientUtilities.ts

```diff
@@ -541,14 +542,23 @@ export async function refundRequestIsValid(
     };
   }
 
-  // Validate relayer and depositId.
+  if (previousIdenticalRequests.gt(0)) {
+    return { valid: false, reason: "Previous identical request exists" };
+  }
+
+  // Validate relayer and depositId. Also check that fill requested refund on same chain that
+  // refund was sent.
   const fill = destSpoke.getFillsForRelayer(relayer).find((fill) => {
     // prettier-ignore
     return (
         fill.depositId === depositId
         && fill.originChainId === originChainId
         && fill.destinationChainId === destinationChainId
+        // Must have requested refund on chain that refund was sent on.
+        && fill.repaymentChainId === repaymentChainId
+        // Must be a full fill to qualify for a refund.
         && fill.amount.eq(amount)
+        && fill.fillAmount.eq(amount)
```

Comment:
This check isn't necessarily required since partial fills are forced by the contracts to take repayment on destination chain but its a safety check?

Reply:
This makes sense to me